### PR TITLE
Support centered settings

### PR DESCRIPTION
**Pull request by yoyo837**
First of all, thank you for your contribution! :-)

Clean #11396

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style. `windows`
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [x] Update API docs for the component.
  * [x] Update/Add demo to demonstrate new feature.
  * [x] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

**Comment by yoyo837** [comment]
@afc163 The ci tests has failed, what did I missed?

File: components/modal/Modal.tsx [hunk 1]
```diff
@@ -188,7 +194,8 @@ export default class Modal extends React.Component<ModalProps, {}> {
 
     return (
       <Dialog
-        {...this.props}
+        {...restProps}
```
Comment (afc163) [inline components/modal/Modal.tsx]: prefixCls is missing in restProps

Comment (afc163) [reply components/modal/Modal.tsx]: image

Comment (afc163) [reply components/modal/Modal.tsx]: Please run `npm test` in your local enviorment to make sure test is OK.

Comment (afc163) [reply components/modal/Modal.tsx]: You can keep `this.props` usage.

File: components/modal/Modal.tsx [hunk 2]
```diff
@@ -188,7 +194,9 @@ export default class Modal extends React.Component<ModalProps, {}> {
...
+        prefixCls={prefixCls}
+        className={classNames({ [`${prefixCls}-centered`]: centered }, className)}
```
Comment (afc163) [inline components/modal/Modal.tsx]: !!centered

File: components/modal/confirm.tsx [hunk 3]
```diff
@@ -32,6 +32,9 @@ const ConfirmDialog = (props: ConfirmDialogProps) => {
   const classString = classNames(
     prefixCls,
     `${prefixCls}-${props.type}`,
+    {
+      [`${prefixCls}-centered`]: props.centered,
```
Comment (afc163) [inline components/modal/confirm.tsx]: !!props.centered,

File: components/modal/Modal.tsx (hunk 1 above)
Comment (yoyo837) [reply components/modal/Modal.tsx]: I want to drop the `centered`.

Comment (afc163) [reply components/modal/Modal.tsx]: 👌

**Pull request merged**